tests/test_servefile.py

```diff
@@ -20,6 +20,23 @@
     connrefused_exc = socket.error
 
 
+def _get_port_from_env(var_name, default):
+    if var_name not in os.environ:
+        return default
+    port = int(os.environ[var_name])
+    if port == 0:
+        # do a one-time port selection for a free port, use it for all tests
+        s = socket.socket()
+        s.bind(('', 0))
+        port = s.getsockname()[1]
+        s.close()
+    return port
+
+
+SERVEFILE_DEFAULT_PORT = _get_port_from_env('SERVEFILE_DEFAULT_PORT', 8080)
+SERVEFILE_SECONDARY_PORT = _get_port_from_env('SERVEFILE_SECONDARY_PORT', 8081)
```

Comment:
My vote for picking less troublesome defaults, e.g. 58080 and 58081. That has a much higher chance to not collide, by default.

Reply:
I see the appeal. Especially for 8081 there is no need for it to be that value. For 8080 on the other hand.. It's the servefile default port. Originally I wanted to test servefile without ever specifying a port and that's how all except one test are written (the one testing the `-p` flag). With this commit I wanted to provide people who cannot run the servefile tests in their intended way to automatically use another port. @grandchild made an excellent case for this, as he wanted to run the tests for his servefile AUR package and that's what also motivated me to implement this feature (thanks @grandchild).

I think to emotionally move away from the "but I want to run servefile on the default port!" I would need another nudge/argument. If I were to do this though in my opinion the proper way would be to always go with autodetection by specifying `0` for both ports, so there would never be a clash.

Reply:
My vote for always using 0-auto-detection then, I suppose.

Reply:
Done.